src/main/java/tech/picnic/jolo/ObjectGraph.java

```diff
@@ -0,0 +1,139 @@
+package tech.picnic.jolo;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.mapping;
+import static tech.picnic.jolo.Util.validate;
+
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.MultimapBuilder;
+import com.google.common.collect.SetMultimap;
+import com.google.common.collect.Table;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.StringJoiner;
+import javax.annotation.Nullable;
+import org.jooq.Record;
+
+/**
+ * Mutable graph of objects loaded by {@link Entity entities}. Objects are linked by {@link IdPair
+ * ID pairs} loaded by {@link Relation relations}.
+ *
+ * @apiNote Access is not synchronized.
+ */
+final class ObjectGraph {
```

Comment:
Can we make this class public?

For r2dbc support we cannot use `Query.collect` but have to manually load records. We use to do this with the `Loader.next`. Which would now become: `loader.accumulator().accept(objectGraph...` 

And `Loader.getList` changes to `loader.finisher().apply(objectGraph)`.

(might be good to document this as well)

Reply:
Can you try out @EnricSala 's suggestion first?

I'd like to avoid extending the public API surface unnecessarily (even though Jolo does not have a compatibility guarantee).

Reply:
The suggestion to just map the flux of r2dbc rows to jooq records and then calling `collect` is actually a lot easier.
No need to make it public.